samples/js/graphiql/src/main.js

```diff
@@ -93,27 +117,28 @@ class Prompt extends React.Component<void, PromptProps, void> {
 
   _handleOnSubmit(e) {
     e.preventDefault();
-    const qs = ['db', 'ds', 'auth']
+    const qs = ['db', 'endpoint', 'ds', 'auth', 'extra']
       .map(k => [k, this.refs[k].value])
       .filter(([, v]) => !!v)
-      .map(([k, v]) => `${k}=${v}`)
+      .map(([k, v]) => `${k}=${encodeURIComponent(v)}`)
       .join('&');
     window.history.pushState({}, undefined, qs === '' || ('?' + qs));
     load();
   }
 }
 
 function graphQLFetcher(graphQLParams) {
-  const url = `${params.db}/graphql/`;
+  const url = `${params.db}/${params.endpoint}`;
 
   const headers = new Headers();
   headers.append('Content-Type', 'application/x-www-form-urlencoded; charset=utf-8');
   if (params.auth) {
     headers.append('Authorization', `Bearer ${params.auth}`);
   }
 
+  const extra = params.extra ? ('&' + params.extra) : '';
```

Comment:
Not what I want to do, the purpose of this is to have non-encoded URI components so you can go `db=foo&something=bar`, which wouldn't be right if it were URI encoded.